### PR TITLE
[CI] Parallelize Train test runners

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

This runs Train tests and examples job in 2 runners in parallel, adjusts timeouts of some Train and AIR tests to match bazel suggestions (for better sharding) and makes minor changes to AIR test names for consistency. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .buildkite/pipeline.ml.yml [hunk 1]
```diff
@@ -1,14 +1,44 @@
...
+      python ./ci/run/bazel_sharding/bazel_sharding.py
+      --exclude_manual
+      --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}"
+      --tag_filters=-gpu_only,-gpu,-minimal,-tune,-needs_credentials
+      python/ray/train/...
+      > test_shard.txt
+    - cat test_shard.txt
+    - bazel test --config=ci $(./ci/run/bazel_export_options)
+      --test_tag_filters=-gpu_only,-gpu,-minimal,-tune,-needs_credentials
+      $(cat test_shard.txt)
```
Comment (krfricke) [inline .buildkite/pipeline.ml.yml:31]: Wondering if we can simplify this to

```suggestion
    - python ./ci/run/bazel_sharding/bazel_sharding.py
      --exclude_manual
      --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}"
      --tag_filters=-gpu_only,-gpu,-minimal,-tune,-needs_credentials
      python/ray/train/... | tee /dev/fd/2 | xargs bazel test --config=ci $(./ci/run/bazel_export_options)
      --test_tag_filters=-gpu_only,-gpu,-minimal,-tune,-needs_credentials
```

Comment (krfricke) [reply .buildkite/pipeline.ml.yml:31]: Actually we should probably just have a script for that. We shouldn't need to pass the test tag filters twice

**Review by krfricke — COMMENTED** [review]
LGTM but let's add a script to simplify calling bazwel with test filters

File: .buildkite/pipeline.ml.yml (hunk 1 above)
Comment (Yard1) [reply .buildkite/pipeline.ml.yml]: @krfricke added the script, PTAL!

**Review by krfricke — APPROVED** [review]
LGTM

**Pull request merged**